From f3acfdbc2b4f14a804d33b062f834c7f32caf143 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 1 Feb 2021 00:49:39 +0100
Subject: [PATCH 1/1] sandbox: mark os_abort() as __noreturn

gcc -fanalyzer needs the information that a function does not return to
provide accurate information.

os_abort() does not return. Mark it accordingly.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/os.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/os.h b/include/os.h
index e192e32d59..8de802cab0 100644
--- a/include/os.h
+++ b/include/os.h
@@ -341,7 +341,7 @@ void os_localtime(struct rtc_time *rt);
 /**
  * os_abort() - raise SIGABRT to exit sandbox (e.g. to debugger)
  */
-void os_abort(void);
+void __noreturn os_abort(void);
 
 /**
  * os_mprotect_allow() - Remove write-protection on a region of memory
-- 
2.29.2

